docker-images/oncoanalyser/assets/run.sh

```diff
@@ -165,79 +177,175 @@ fi
 
-instance_id=$(curl -s http://169.254.169.254/latest/meta-data/instance-id)
-association_id=$(
-  aws ec2 describe-iam-instance-profile-associations | \
-    jq -r '
-      .IamInstanceProfileAssociations[] |
-        select(.InstanceId == "'${instance_id}'") |
-        .AssociationId
+## SSM Parameter functions
+get_ssm_parameter_value(){
+  aws ssm get-parameter \
+    --name "$1" \
+    --output json |
+  jq '.Parameter | .Value'
+}
+
+get_cache_bucket_from_ssm(){
+  get_ssm_parameter_value "/oncoanalyser/nxf/cache_bucket"
+}
+
+get_cache_path_from_ssm(){
+  get_ssm_parameter_value "/oncoanalyser/nxf/cache_path"
+}
+
+get_dest_bucket_from_ssm(){
+  get_ssm_parameter_value "/oncoanalyser/nxf/staging_bucket"
+}
+
+get_dest_prefix_from_ssm(){
+  get_ssm_parameter_value "/oncoanalyser/nxf/staging_prefix"
+}
+
+get_hmf_refdata_from_ssm(){
+  get_ssm_parameter_value "/oncoanalyser/reference_data/hmf"
+}
+
+get_hmf_virusbreakend_db_from_ssm(){
+  get_ssm_parameter_value "/oncoanalyser/reference_data/virusbreakend"
+}
+
+get_hmf_genomes_path_from_ssm(){
+  get_ssm_parameter_value "/oncoanalyser/reference_data/genomes"
+}
+
+get_batch_instance_role_name_from_ssm(){
+  get_ssm_parameter_value "/oncoanalyser/iam/batch-instance-role-name"
+}
+
+
+get_ica_secrets_portal_provider_arn(){
+  aws lambda get-function \
+    --function-name "IcaSecretsPortalProvider" \
+    --output json | \
+  jq --raw-output \
+    '.Configuration | .FunctionArn'
+}
+
+get_batch_instance_role_arn(){
+  aws iam get-role \
+    --role-name "$(get_batch_instance_role_name_from_ssm)" \
+    --output json | \
+  jq --raw-output \
+   '
+     .Role.Arn
+   '
+}
+
+get_batch_instance_profile_name(){
+  aws iam list-instance-profiles-for-role \
+    --role-name "$(get_ssm_parameter_value "/oncoanalyser/iam/batch-instance-profile-name")" \
+    --output json | \
+  jq --raw-output \
     '
-)
-aws ec2 replace-iam-instance-profile-association 1>/dev/null \
-  --association-id ${association_id} \
-  --iam-instance-profile Name='OncoanalyserStack-OncoanalyserTaskBatchInstanceProfile-Eq6HgOiK5G71'
+      .InstanceProfiles |
+      map(.InstanceProfileName) |
+      .[]
+    '
+}
+
 
-if [[ ! -z ${resume_nextflow_dir:-} ]]; then
-  aws s3 sync \
-    --no-progress \
-    ${resume_nextflow_dir}/ ./.nextflow/
-fi
 
 stage_gds_fp() {
   gds_fp=${1}
   gds_dp=${gds_fp%/*}/
 
-  dst_bucket=umccr-temp-dev
-  dst_key_base=stephen/gds_staging_dev
+  dst_bucket="$(get_dest_bucket_from_ssm)"
+  dst_key_base="$(get_dest_prefix_from_ssm)"
 
   dst_dp=${dst_bucket}/${dst_key_base}
   dst_fp=${dst_dp}/${gds_fp##*/}
 
   echo s3://${dst_fp}
 
   if [[ -z "${ICA_ACCESS_TOKEN:-}" ]]; then
-    AWS_DEFAULT_REGION=ap-southeast-2 aws lambda invoke \
-      --function-name arn:aws:lambda:ap-southeast-2:472057503814:function:IcaSecretsPortalProvider \
+    AWS_REGION=ap-southeast-2 aws lambda invoke \
+      --function-name "$(get_ica_secrets_portal_provider_arn)" \
       response.json 1>/dev/null
```

Comment:
The IcaSecretsPortalProvider Lambda only exists in prod and has been used to grant read permissions on the GDS production volume for oncoanalyser running dev. Having the CodePipeline that allows us to properly deploy into stg/prod means that we no longer need that work around and instead can use the IcaSecretsPortal token from Secrets Manager.

Leaving it open to you whether you'd like to make that change, otherwise I'll update outside of this PR.

Reply:
Updated in 7acc9c13da2d7887fd0065cbb5f7971d1f22d4dc to use the secret rather than the lambda function to collect the access token